src/main/java/jcmp/DocCompare.java

```diff
@@ -394,6 +393,9 @@ public int hashCode() {
 
         @Override
         public boolean equals(Object x) {
+            if (!(x instanceof DefaultIdentity)) {
```

Comment:
Sure, but isn't an if going to be less expensive than a class cast exception being created, thrown, then caught?  Could extend the if to check for null then get rid of the try/catch completely.